commonutils/src/main/java/io/intrepid/commonutils/AndroidStringUtils.java

```diff
@@ -89,4 +97,54 @@ public static Spanned fromHtml(String source, int flags) {
             return Html.fromHtml(source);
         }
     }
+
+    /**
+     * Attempts to parse the given String to a Uri.
+     * This method exists because standard pattern matching functions (such as {@link Patterns#WEB_URL} treat addresses
+     * that omit a protocol (i.e. beginning with 'www') as valid.  However, if you try to actually use this Uri
+     * (e.g. calling CustomTabsIntent.launchUrl(Context, Uri)), it won't work.
+     * <p>
+     * This method will look for the case where the URL string is missing the protocol, and will prepend it if necessary.
+     *
+     * @param inputUri The String that you want to try parsing as a Uri
+     * @param scheme   The scheme (e.g. 'http://', 'https://', 'ftp://') to prepend to this inputUri if it did not have a scheme
+     * @return A valid, well-formed Uri, or null if the inputUri parameter was not a valid URI
+     */
+    @Nullable
+    public static Uri parseUriFromString(@NonNull String inputUri, @NonNull String scheme) {
+        try {
+            URI uri = new URI(inputUri);
+            if (uri.getScheme() == null) {
+                uri = new URI(scheme + inputUri);
+            }
+            return Uri.parse(uri.toString());
+        } catch (Exception e) {
+            Log.e(AndroidStringUtils.class.getSimpleName(), "Could not parse URI", e);
```

Comment:
Personally I'm much more a fan of checked exceptions for parsing code because then you KNOW the caller has to deal with it.  I'd prefer rethrowing a `MalformedURLException` or whatnot here, but I'll leave it to you to decide.